text/0474-event-bridge-scheduler-l2.md

```diff
@@ -0,0 +1,481 @@
+2. **Targets**: A target is an API operation that EventBridge Scheduler calls on your behalf every time your schedule runs.  Targets can include Amazon EC2 instances, AWS Lambda functions, Kinesis streams, Amazon ECS tasks, Step Functions state machines, Amazon SNS topics, Amazon SQS queues, Amazon CloudWatch LogGroups, and built-in targets. A target receives events in JSON format.
+3. **Schedule Group**: Scheduling groups help you organize your schedules. Scheduling groups support tags that you can use for cost allocation, access control, and resource organization. When creating a new schedule, you can add it to a scheduling group.
+
+This module is part of the [AWS Cloud Development Kit](https://github.com/aws/aws-cdk) project. It allows you to define Event Bridge Schedules.
+
+
+## Defining a schedule 
+
+```ts
+
+const target = new targets.LambdaInvoke(props.func, {
+    input: ScheduleTargetInput.fromObject({
+    "payload": "useful"
+    })
+});
+    
+const schedule = new Schedule(this, 'Schedule', {
+    schedule: ScheduleExpression.rate(Duration.minutes(10)),
+    target,
+    description: 'This is a test schedule that invokes lambda function every 10 minutes.',
+});
+```
+
+### Schedule Expressions
+
+Both rate-based and cron-based schedules are recurring schedules. You configure each recurring schedule type using a schedule expression for the type of schedule you want to configure, and specifying a time zone in which EventBridge Scheduler evaluates the expression.
+
+```ts
+const rateBasedSchedule = new Schedule(this, 'Schedule', {
+    schedule: ScheduleExpression.rate(Duration.minutes(10)),
+    target,
+    description: 'This is a test rate-based schedule',
+});
+
+const cronBasedSchedule = new Schedule(this, 'Schedule', {
+    schedule: ScheduleExpression.cron({ day: '20', month: '11' }),
+    target,
+    scheduleTimeZone: 'America/New_York'
+    description: 'This is a test cron-based schedule in New York timezone',
+});
+```
+
+A one-time schedule is a schedule that invokes a target only once. You configure a one-time schedule when by specifying the time, date, and time zone in which EventBridge Scheduler evaluates the schedule.
+
+```ts
+const oneTimeSchedule = new Schedule(this, 'Schedule', {
+    schedule: ScheduleExpression.at(new Date(2022, 10, 20, 19, 20, 23)),
+    target,
+    scheduleTimeZone: 'America/New_York'
+    description: 'This is a one-time schedule in New York timezone',
+});
+```
+
+### Grouping Schedules
+
+When creating a new schedule, you can add it to a scheduling group:
+
+```ts
+const group = new Group(this, "Group", {
+    groupName: "MyGroup"
```

Comment:
Are there going to be more props to a group? Or is it simply a name.

Also, should we add APIs like `group.addSchedules(schedules: Schedule[])`?

Reply:
> Are there going to be more props to a group? Or is it simply a name.

By them you mean the owners of AWS EB Scheduler service? I do not know, how can we check that? 

> Also, should we add APIs like group.addSchedules(schedules: Schedule[])?

Sounds like a good idea! 

A question for implementation - currently, we create `CfnSchedule` in a constructor of `Schedule` and pass `groupName` parameter to `CfnSchedule`. How can we delay creation of `CfnSchedule` so that we can change the `groupName` parameter dynamically? Is there a good example of that behaviour somewhere in CDK lib already?

Reply:
Yes. I think you're asking for Lazy :). We use it all over the place. [here](https://github.com/aws/aws-cdk/blob/cabff71139bf6a3b8ecd4cdf4a8dea093e67b8e6/packages/%40aws-cdk/aws-events/lib/rule.ts#L112) is an example.